src/openapi/streaming/subscription-queue.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Queue (FIFO) of pending subscription actions. Allows for easier abstraction of managing list of pending actions.
+ * In addition to standard queue implementation, provides:
+ *  - merging consecutive duplicated actions
+ *  - merging of uneccessary combinations: ACTION_SUBSCRIBE action before ACTION_UNSUBSCRIBE, is merged to ACTION_UNSUBSCRIBE.
+ *
+ * @module saxo/openapi/streaming/subscription queue
+ * @ignore
+ */
+
+import {
+    ACTION_SUBSCRIBE,
+    ACTION_UNSUBSCRIBE
+} from './subscription-actions';
+
+/**
+ * Queue (FIFO) for storing pending subscription actions.
+ * @param maxSize {Number} - Maximum queue size. Defaults to 2, which works best for current needs.
+ * @constructor
+ */
+function SubscriptionQueue(maxSize = 2) {
+    this.queue = [];
+    this.maxSize = maxSize;
+}
+
+/**
+ * Add pending action to a queue and making sure that we remove unecessary actions like:
+ * - duplicates.
+ * - ACTION_SUBSCRIBE action before ACTION_UNSUBSCRIBE.
+ *
+ * @param {Number} action - action to add to the queue.
+ */
+SubscriptionQueue.prototype.enqueue = function(action) {
+    this.queue.push(action);
+
+    console.log('[enque] add action: ', action, ' MaxSize: ', this.maxSize);
```

Comment:
Added jshint rule to disallow console usage. Also removed existing usage as it was found int print-task.

Created separate PR for it as this one will probably take some time.
https://github.com/SaxoBank/openapi-clientlib-js/pull/19